app/models/ranking/popular_posts.rb

```diff
@@ -0,0 +1,54 @@
+class Ranking::PopularPosts
+  def ranked_posts(site)
+    @site = site
+    pageviews_per_post.map do |post|
+      next unless valid?(post)
```

Comment:
How about
`posts.select {|post| valid?(post) }.map {|p| {post_id: post_id} }.sort_by(&:score)` ?

We can use `select` instead of `valid?` and `compact` and
`sort_by` instead of `sort! {}`